ballerina/tests/subscriber_with_readonly_method_params_test.bal

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/log;
+import ballerina/test;
+import ballerina/mime;
+import ballerina/http;
+
+@SubscriberServiceConfig {}
+service /subscriber on new Listener(9104) {
+    isolated remote function onSubscriptionValidationDenied(readonly & SubscriptionDeniedError msg) returns Acknowledgement? {
+        return ACKNOWLEDGEMENT;
+    }
+
+    isolated remote function onSubscriptionVerification(readonly & SubscriptionVerification msg)
+                        returns SubscriptionVerificationSuccess|SubscriptionVerificationError {
+        if (msg.hubTopic == "test1") {
```

Comment:
```suggestion
       boolean isReadOnly = SubscriptionVerification is readonly;
        if (msg.hubTopic == "test1") {
```
Shall we check the readOnly state in a test case to see the correct functionality?

Reply:
Fixed with [981e761726a46c53c607bd2ed14dab3713ce66a5](https://github.com/ballerina-platform/module-ballerina-websub/pull/500/commits/981e761726a46c53c607bd2ed14dab3713ce66a5)